### PR TITLE
Dịch "softmax-regression-scratch_vn" - Phần 3

### LINKED ISSUE #769: Dịch "softmax-regression-scratch_vn" - Phần 3
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_linear-networks/softmax-regression-scratch_vn.md#L152-L215

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by lamngoctam**
1st draft - Không thấy ví dụ hàm `pick` ở cuối phần 3.

## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #769 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 1]
```diff
@@ -178,14 +180,15 @@ def net(X):
...
-## *dịch tiêu đề phía trên*
+## Hàm mất mát
 
 <!--
 Next, we need to implement the cross-entropy loss function, introduced in :numref:`sec_softmax`.
 This may be the most common loss function in all of deep learning because, at the moment, classification problems far outnumber regression problems.
 -->
 
-*dịch đoạn phía trên*
+Tiếp đến chúng ta cần thực thi hàm mất mát cross-function được giới thiệu ở :numref:`sec_softmax`.
```
Comment (lamngoctam) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Tiếp đến chúng ta cần thực thi hàm mất mát cross-function đã được giới thiệu ở :numref:`sec_softmax`.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 2]
```diff
@@ -194,7 +197,10 @@ we can use the `pick` function which allows us to easily select the appropriate
 Below, we illustrate the `pick` function on a toy example, with $3$ categories and $2$ examples.
 -->
 
-*dịch đoạn phía trên*
+Cần ghi nhớ rằng cross-entropy có chứa negative log likelihood của xác suất dự đoán được gán cho nhãn thực $-\log P(y \mid x)$.
+Thay vì sử dụng lặp lại sự tiên doán với vòng lặp `for` trong Python (kém hiệu quả),
```
Comment (lamngoctam) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Thay vì sử dụng lặp lại sự tiên đoán bằng vòng lặp `for` trong Python (kém hiệu quả),
```

Comment (lamngoctam) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Thay vì sử dụng lặp lại sự tiên đoán bằng vòng lặp `for` trong Python (có xu hướng kém hiệu quả),
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 3]
```diff
@@ -205,7 +211,7 @@ y_hat[[0, 1], [0, 2]]
 Now we can implement the cross-entropy loss function efficiently with just one line of code.
 -->
 
-*dịch đoạn phía trên*
+Bây giờ chúng ta có thể thục thi hàm mất mát cross-entropy đơn giản với một dòng lệnh.
```
Comment (lamngoctam) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Bây giờ chúng ta có thể thực thi hàm mất mát cross-entropy hiệu quả hơn với một dòng lệnh.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 4]
```diff
@@ -194,7 +197,10 @@ we can use the `pick` function which allows us to easily select the appropriate
 Below, we illustrate the `pick` function on a toy example, with $3$ categories and $2$ examples.
 -->
 
-*dịch đoạn phía trên*
+Cần ghi nhớ rằng cross-entropy có chứa negative log likelihood của xác suất dự đoán được gán cho nhãn thực $-\log P(y \mid x)$.
```
Comment (rootonchair) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Nhắc lại rằng entropy chéo lấy kết quả là hàm đối log hợp lý của xác suất dự đoán được gán cho nhãn thực $-\log P(y \mid x)$.
```
Phần này là phần kế tiếp của phần softmax regression nên nhiều từ đã được định nghĩa bên đó

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 5]
```diff
@@ -194,7 +197,10 @@ we can use the `pick` function which allows us to easily select the appropriate
...
+Thay vì sử dụng lặp lại sự tiên đoán bằng vòng lặp `for` trong Python (có xu hướng kém hiệu quả),
```
Comment (rootonchair) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Thay vì lặp qua các dự đoán của mô hình bằng vòng lặp `for` trong Python (có xu hướng kém hiệu quả),
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 6]
```diff
@@ -194,7 +197,10 @@ we can use the `pick` function which allows us to easily select the appropriate
...
+chúng ta có thể sử dụng hàm `pick` mà cho phép ta chọn lựa dễ dàng các thuật ngữ thích hợp từ ma trận của các biến đầu vào softmax.
```
Comment (rootonchair) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
chúng ta có thể sử dụng hàm `pick` mà cho phép ta chọn lựa dễ dàng các phần tử thích hợp từ ma trận của các biến softmax đầu vào.
```

Bạn để mỗi câu là một dòng nhé

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 7]
```diff
@@ -194,7 +197,10 @@ we can use the `pick` function which allows us to easily select the appropriate
...
+Sau đây hàm `pick` được sử dụng như một ví dụ đơn giản với ma trận $2$ hàng $3$ cột.
```
Comment (rootonchair) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Dưới đây, hàm `pick` được sử dụng như một ví dụ đơn giản với ma trận $2$ hàng $3$ cột.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 8]
```diff
@@ -205,7 +211,7 @@ y_hat[[0, 1], [0, 2]]
 Now we can implement the cross-entropy loss function efficiently with just one line of code.
 -->
 
-*dịch đoạn phía trên*
+Bây giờ chúng ta có thể thực thi hàm mất mát cross-entropy hiệu quả hơn với một dòng lệnh.
```
Comment (rootonchair) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Bây giờ chúng ta có thể lập trình hàm mất mát entropy chéo hiệu quả hơn với một dòng lệnh.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 9]
```diff
@@ -159,15 +159,17 @@ X_prob, X_prob.sum(axis=1)
...
 
 <!--
 Now that we have defined the softmax operation, we can implement the softmax regression model.
 The below code defines the forward pass through the network.
 Note that we flatten each original image in the batch into a vector with length `num_inputs` with the `reshape` function before passing the data through our model.
 -->
 
-*dịch đoạn phía trên*
+Bây giờ chúng ta định nghĩa phép tính softmax, sau đó chúng ta có thể thực thi mô hình hồi quy softmax.
+Đoạn mã sau xác định sự chuyển tiếp thông qua mạng.
```
Comment (rootonchair) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]: `forward pass` dịch thế này mình thấy vẫn chưa ổn. Chắc cần thảo luận thêm

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 10]
```diff
@@ -159,15 +159,17 @@ X_prob, X_prob.sum(axis=1)
...
+## Mô hình
 
 <!--
 Now that we have defined the softmax operation, we can implement the softmax regression model.
 The below code defines the forward pass through the network.
 Note that we flatten each original image in the batch into a vector with length `num_inputs` with the `reshape` function before passing the data through our model.
 -->
 
-*dịch đoạn phía trên*
+Bây giờ chúng ta định nghĩa phép tính softmax, sau đó chúng ta có thể thực thi mô hình hồi quy softmax.
```
[the remark this answers is not in the document]
Comment (lamngoctam) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Thanks for your all comments!!!

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 11]
```diff
@@ -178,14 +180,15 @@ def net(X):
...
+## Hàm mất mát
 
 <!--
 Next, we need to implement the cross-entropy loss function, introduced in :numref:`sec_softmax`.
 This may be the most common loss function in all of deep learning because, at the moment, classification problems far outnumber regression problems.
 -->
 
-*dịch đoạn phía trên*
+Tiếp đến chúng ta cần lập trình hàm mất mát entropy chéo đã được giới thiệu ở :numref:`sec_softmax`.
+Đây có lẽ là hàm mất mát thông dụng nhất trong tất cả nghiên cứu về học sâu vì hiện nay số lượng bài toán phân loại vượt trội hơn số lượng bài toán về hồi quy.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Đây có lẽ là hàm mất mát thông dụng nhất trong nghiên cứu về học sâu vì hiện nay số lượng bài toán phân loại vượt trội hơn số lượng bài toán hồi quy.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 12]
```diff
@@ -205,7 +211,7 @@ y_hat[[0, 1], [0, 2]]
 Now we can implement the cross-entropy loss function efficiently with just one line of code.
 -->
 
-*dịch đoạn phía trên*
+Bây giờ chúng ta có thể lập trình hàm mất mát entropy chéo hiệu quả hơn với một dòng lệnh.
```
Comment (tanduong) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Bây giờ chúng ta có thể lập trình hàm tính mất mát entropy chéo chạy nhanh hơn chỉ với một dòng lệnh.
```

Comment (lamngoctam) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: @tamnguyenvan dịch là `hàm tính mất mát entropy chéo` nge không thuận vì gốc từ là danh từ, không phải là tính từ. 
Mình đồng ý với cụm dịch `chạy nhanh hơn` có ý nghĩa xác thực và chi tiết hơn so với từ `hiệu quả`

Comment (tamnguyenvan) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Bạn tag nhầm mình hả :))

Comment (tiepvupsu) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Mình vote giữ nguyên như @lamngoctam nhé. `cross-entropy loss function` cả cụm dịch là `hàm mất mát entropy chéo`, còn chữ `tính` mình cho là không cần thiết.

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 13]
```diff
@@ -549,7 +555,7 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 3 -->
-*
+* Lâm Ngọc Tâm
```
Comment (lamngoctam) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
* Lâm Ngọc Tâm
* Phạm Hồng Vinh
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 14]
```diff
@@ -159,15 +159,17 @@ X_prob, X_prob.sum(axis=1)
...
 
 <!--
 Now that we have defined the softmax operation, we can implement the softmax regression model.
 The below code defines the forward pass through the network.
 Note that we flatten each original image in the batch into a vector with length `num_inputs` with the `reshape` function before passing the data through our model.
 -->
 
-*dịch đoạn phía trên*
+Bây giờ chúng ta đã định nghĩa hàm softmax, chúng ta có thể bắt đầu lập trình mô hình hồi quy softmax.
+Đoạn mã sau xác định sự chuyển tiếp thông qua mạng.
```
Comment (lamngoctam) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Đoạn mã sau định nghĩa lượt truyền xuôi thông qua mạng.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 12 above)
Comment (lamngoctam) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Vậy mình vẫn dịch là `hàm mất mát entropy chéo` như comment của @tiepvupsu.
@tamnguyenvan sorry, mình tag nhầm ^^

Comment (lamngoctam) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Bây giờ chúng ta có thể lập trình hàm mất mát entropy chéo chạy nhanh hơn chỉ với một dòng lệnh.
```
